sentry-spring/src/main/java/io/sentry/spring/SentryUserProviderEventProcessor.java

```diff
@@ -15,6 +15,10 @@
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
+/**
+ * Deprecated: instead of using event processor for setting user on events and transactions, user
+ * should be set on the scope using {@link SentryUserFilter}.
+ */
```

Comment:
I don't follow this comment, this filter should be only to address the missing user from transactions and not events, events should not be affected otherwise its a breaking change experience even for events or am I missing something?

Also, we should annotate with `@Deprecated`

Reply:
This means that using this filter alone solves the problem of setting user on events and transactions. We keep it to not break codebases that do not add the filter. In long term: users should use filter and not this event processor.

Reply:
ah ok, `SentryInitBeanPostProcessor` still adds it, so `SentryUserProviderEventProcessor` will get executed for the Spring integration/events not causing a breaking change.